pkg/upid/types.go

```diff
@@ -0,0 +1,142 @@
+const (
+	PlatformIDTypeNotSet          uint32 = 0 // UPID_OEM_PLATFORM_ID_TYPE_NOT_SET
+	PlatformIDTypeBinary          uint32 = 1 // UPID_OEM_PLATFORM_ID_TYPE_BINARY
+	PlatformIDTypePrintableString uint32 = 2 // UPID_OEM_PLATFORM_ID_TYPE_PRINTABLE_STRING
+)
+
+// UPID Feature States
+const (
+	FeatureStateDisabled uint8 = 0x00
+	FeatureStateEnabled  uint8 = 0x01
+)
+
+// Intel UPID Status Codes (from Intel UPID Attestation SDK Documentation)
+const (
+	StatusSuccess                  uint8 = 0
+	StatusFeatureNotSupported      uint8 = 1
+	StatusInvalidInputParameter    uint8 = 2
+	StatusInternalError            uint8 = 3
+	StatusNotAllowedAfterEOP       uint8 = 4
+	StatusNotAllowedAfterManufLock uint8 = 5
+	StatusMaxCountersExceeded      uint8 = 6
+	StatusInvalidState             uint8 = 7
+	StatusReserved2                uint8 = 8
+	StatusNotAllowedAfterCBD       uint8 = 9
+)
+
+// Common errors
+var (
+	ErrUPIDNotSupported         = errors.New("intel UPID is not supported on this platform")
+	ErrUPIDNotEnabled           = errors.New("intel UPID feature is not enabled")
+	ErrUPIDNotProvisioned       = errors.New("intel UPID is not provisioned on this platform")
+	ErrInvalidResponse          = errors.New("invalid response from UPID MEI client")
+	ErrConnectionFailed         = errors.New("failed to connect to UPID MEI client")
+	ErrCommandFailed            = errors.New("UPID command failed")
+	ErrInsufficientPriv         = errors.New("insufficient privileges to access UPID")
+	ErrFeatureNotSupported      = errors.New("UPID feature not supported")
+	ErrInvalidInputParameter    = errors.New("invalid input parameter")
+	ErrInternalError            = errors.New("UPID internal error")
+	ErrNotAllowedAfterEOP       = errors.New("operation not allowed after end of post")
+	ErrInvalidState             = errors.New("UPID is in invalid state or disabled after EOP/EOM")
+	ErrNotAllowedAfterManufLock = errors.New("operation not allowed after manufacturing lock")
+	ErrMaxCountersExceeded      = errors.New("maximum counters exceeded")
+	ErrNotAllowedAfterCBD       = errors.New("operation not allowed after core BIOS done")
+)
+
+// UPID represents the Intel Unique Platform Identifier
+type UPID struct {
+	// Full 64-byte UPID
+	Raw []byte `json:"raw"`
+	// OEMPlatformID is the first 32 bytes
+	OEMPlatformID []byte `json:"oemPlatformID"`
+	// HWSerialNum is the second 32 bytes
+	HWSerialNum []byte `json:"hwSerialNum"`
+}
+
+// Interface defines the operations for Intel UPID
+type Interface interface {
+	// GetUPID retrieves the Intel UPID from the platform
+	// Returns the UPID struct or an error if not supported/enabled
+	GetUPID() (*UPID, error)
+
+	// IsSupported checks if Intel UPID is supported on this platform
+	IsSupported() bool
+
+	// Close releases any resources held by the UPID client
+	Close() error
+}
+
+// NewUPID creates a new UPID structure from raw 64-byte data
+func NewUPID(raw []byte) (*UPID, error) {
```

Comment:
This seems to be used only by the linux implementation. In that case no need to have this in the types file.